Fix jenkins common role  TE-2210

jzoldak opened the pull request:
* Add logging for jenkins app, with logrotate
* Fix the user:group of installed files
* Specify the uid:gid for jenkins to match the attached data volume
* Configure jenkins.war expansion into /var/cache
* Add Splunkforwarder installation and configuration

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

feanil commented on playbooks/roles/jenkins_common/defaults/main.yml:
```diff
@@ -1,7 +1,9 @@
 jenkins_common_user: jenkins
-jenkins_common_group: edx
+jenkins_common_user_uid: 1002
```
Per our conversation yesterday, theses sholud default to null and the role should just pick the next available UIDs.  However if we do explicitly set these, than the role should try to use them.

jzoldak commented:
@feanil @estute 
I pushed a commit that will default to taking the next gid / uid for the jenkins group / user, and allow us to override in order to match up with those on our data volume.

It's not elegant. Setting the value to None did not work in my testing. Instead I had to add when clauses with opposing conditions. :\

Here's the output from provisioning:
```
------------ output when not specifying gid & uid ------------
TASK [jenkins_common : Create jenkins group with specified gid] ****************
skipping: [test_jenkins]

TASK [jenkins_common : Create jenkins group] ***********************************
changed: [test_jenkins]

TASK [jenkins_common : Create the jenkins user with specified uid and add to the group] ***
skipping: [test_jenkins]

TASK [jenkins_common : Create the jenkins user and add to the group] ***********
changed: [test_jenkins]

------------ output when specifying gid & uid ------------
TASK [jenkins_common : Create jenkins group with specified gid] ****************
changed: [test_jenkins]

TASK [jenkins_common : Create jenkins group] ***********************************
skipping: [test_jenkins]

TASK [jenkins_common : Create the jenkins user with specified uid and add to the group] ***
changed: [test_jenkins]

TASK [jenkins_common : Create the jenkins user and add to the group] ***********
skipping: [test_jenkins]
```

michaelyoungstrom commented on playbooks/roles/jenkins_common/defaults/main.yml:
```diff
@@ -1,7 +1,7 @@
+jenkins_common_group: jenkins
 jenkins_common_home: /var/lib/jenkins
-jenkins_common_config_path: /init-configs
+jenkins_common_config_path: /var/lib/jenkins/init-configs
```
Might be a good idea to make this "{{ jenkins_common_home }}/init-configs" (same idea for git home on line 24)

jzoldak commented on playbooks/roles/jenkins_common/templates/etc/logrotate.d/jenkins_log.j2:
```diff
@@ -0,0 +1,11 @@
+# Put in place by ansible
+
+/var/log/jenkins/*jenkins.log {
```
reviewer note: I copied the config verbatim from the official jenkins packaging repo: https://github.com/jenkinsci/packaging/blob/master/deb/build/debian/jenkins.logrotate

jzoldak commented on playbooks/roles/jenkins_build/defaults/main.yml:
```diff
@@ -1,3 +1,5 @@
+build_jenkins_user_uid: 1002
```
Reviewer note: this is so that when we attach the data volume to the instance, the user and group permissions will be correct.
